pkg/util/env/env.go

```diff
@@ -0,0 +1,134 @@
+package env
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/fieldpath"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+type ResourceStore struct {
+	SecretStore    map[string]*kapi.Secret
+	ConfigMapStore map[string]*kapi.ConfigMap
+}
+
+func NewResourceStore() *ResourceStore {
+	return &ResourceStore{
+		SecretStore:    make(map[string]*kapi.Secret),
+		ConfigMapStore: make(map[string]*kapi.ConfigMap),
+	}
+}
+
+func GetEnvVarSecretRefValue(client kclientset.Interface, namespace string, store *ResourceStore, secretSelector *kapi.SecretKeySelector) (string, error) {
```

Comment:
this function has nothing to do w/ EnvVar so don't include it in the name.  find some other way to disambiguate it from the private function.